gateway/app/models/invite.py

```diff
@@ -0,0 +1,186 @@
+from datetime import datetime
+from enum import StrEnum
+from typing import Annotated, Optional, Self
+
+from app.generated.invite_service.invite_service_pb2 import (
+    GrpcInvite,
+)
+from app.generated.invite_service.invite_service_pb2 import (
+    InviteStatus as GrpcInviteStatus,
+)
+from app.validators import str_special_characters_validator
+
+from pydantic import AfterValidator, BaseModel, Field
+from pytz import utc
+
+
+class InviteStatus(StrEnum):
+    """
+    Invite status enum
+
+    Methods
+    -------
+    static from_proto(proto)
+        Get invite status instance from proto invite status
+    to_proto()
+        Get proto invite status from invite status instance
+
+    """
+
+    PENDING = "PENDING"
+    """Invite pending approval"""
+    ACCEPTED = "ACCEPTED"
+    """Invite accepted"""
+    REJECTED = "REJECTED"
+    """Invite rejected"""
+
+    @classmethod
+    def from_proto(cls, proto: GrpcInviteStatus) -> Self:
+        """
+        Get invite status instance from proto invite status
+
+        Parameters
+        ----------
+        proto : GrpcInviteStatus
+            Proto invite status
+
+        Returns
+        -------
+        InviteStatus
+            Invite status instance
+
+        """
+        match proto:
+            case GrpcInviteStatus.PENDING:
+                return cls("PENDING")
+            case GrpcInviteStatus.ACCEPTED:
+                return cls("ACCEPTED")
+            case GrpcInviteStatus.REJECTED:
+                return cls("REJECTED")
+
+        raise ValueError("Unknown invite status")
+
+    def to_proto(self) -> GrpcInviteStatus:
+        """
+        Get proto invite status from invite status instance
+
+        Returns
+        -------
+        GrpcInviteStatus
+            Proto invite status
+
+        """
+        match self:
+            case InviteStatus.PENDING:
+                return GrpcInviteStatus.PENDING
+            case InviteStatus.ACCEPTED:
+                return GrpcInviteStatus.ACCEPTED
+            case InviteStatus.REJECTED:
+                return GrpcInviteStatus.REJECTED
+
+        raise ValueError("Unknown invite status")
+
+
+class Invite(BaseModel):
+    """
+    Invite data class
+
+    Attributes
+    ----------
+    id : str
+        Id of the invite
+    event_id : str
+        Id of the related event
+    author_id : str
+        Id of the author of the invite
+    invitee_id : str
+        Id of the invitee
+    status : InviteStatus
+        Invite status
+    created_at : datetime
+        Time when the invite was created
+    deleted_at : Optional[datetime]
+        Time when invite was deleted
+
+    Methods
+    -------
+    static from_proto(proto)
+        Get invite instance from proto invite
+    to_proto()
+        Get proto invite from invite instance
+
+
+    """
+
+    id: Annotated[
+        str, Field("", min_length=1), AfterValidator(str_special_characters_validator)
```

Comment:
If It's uuid, then set uuid type https://docs.pydantic.dev/2.0/usage/types/uuids/